config/webpack/terra-dev-site.webpack.config.js

```diff
@@ -1,32 +1,98 @@
+*/
+const addAlias = (acc, name, location, hotReloading, { dist, source }, production) => {
+  if (!production && hotReloading) {
+    acc[path.join(name, dist)] = path.join(location, source);
+  }
+  // Always place the more specific aliases above the less specific aliases.
+  acc[name] = location;
+};
+
+/**
+* Alias the current package. This is so you can reference examples as if they are external packages.
+*/
+const aliasCurrentPackage = (packageName, processPath, hotReloading, webpackAliasOptions = {}, production) => {
+  const alias = {};
+  addAlias(alias, packageName, processPath, hotReloading, webpackAliasOptions, production);
+  return alias;
+};
+
+/**
+* Aliases all mono-repo packages. This ensures the correct module is hit if the site is hosting an item used to create itself (ouroboros).
+*/
+const aliasMonoRepoPackages = (hotReloading, monoRepo, webpackAliasOptions = {}, production) => {
+  // If no package directory is found, do nothing.
+  if (!fs.existsSync(monoRepo.packages)) {
+    return {};
+  }
+  // For each non hidden directory, create an alias.
+  return fs.readdirSync(monoRepo.packages).reduce((acc, packageName) => {
+    // ignore any hidden files
+    if (packageName[0] !== '.') {
+      addAlias(acc, packageName, path.join(monoRepo.packages, packageName), hotReloading, webpackAliasOptions, production);
+    }
+    return acc;
+  }, {});
+};
 
-const devSiteConfig = () => {
+/**
+* Generates the file representing app name configuration.
+*/
+const devSiteConfig = (env = {}, argv = {}) => {
+  const production = argv.p;
   const processPath = process.cwd();
-  /* Get the root path of a mono-repo process call */
+  const verbose = env.verboseGenerateAppConfig;
+
+  // Get the root path of a mono-repo process call
   const rootPath = processPath.includes('packages') ? processPath.split('packages')[0] : processPath;
 
-  /* Get the site configuration to add as a resolve path */
-  let devSiteConfigPath = path.resolve(path.join(rootPath, 'dev-site-config'));
-  const customSiteConfigPath = path.join(devSiteConfigPath, 'site.config.js');
-  if (!isFile(customSiteConfigPath)) {
-    devSiteConfigPath = path.join('lib', 'config');
+  // Get the site configuration to add as a resolve path
+  const devSiteConfigPath = path.resolve(path.join(rootPath, 'dev-site-config'));
+
+  // Get default site config.
+  const siteConfig = loadSiteConfig();
+
+  // Generate the files need to spin up the site.
+  generateAppConfig(siteConfig, production, verbose);
+
+  // Is hot reloading enabled?
+  const { hotReloading, monoRepo, webpackAliasOptions } = siteConfig;
+
+  // Get the default package name.
+  const packageName = siteConfig.npmPackage.name;
+
+  // Setup auto aliases.
+  const alias = {
+    ...aliasMonoRepoPackages(hotReloading, monoRepo, webpackAliasOptions, production),
+    ...aliasCurrentPackage(packageName, processPath, hotReloading, webpackAliasOptions, production),
```

Comment:
Does this need to be rootPath, not processPath to correctly run package-level tests?

Reply:
I need to actually remove rootPath. Cleverly we'll build a subset site, just for the package being tested.